app/src/main/java/com/android/unio/ui/association/SaveAssociation.kt

```diff
@@ -55,39 +75,78 @@ import kotlinx.coroutines.launch
  * @param navigationAction [NavigationAction] that handles navigation.
  */
 @Composable
-fun EditAssociationScreen(
+fun SaveAssociationScreen(
     associationViewModel: AssociationViewModel,
     navigationAction: NavigationAction,
+    isNewAssociation: Boolean
 ) {
   val context = LocalContext.current
 
   val association by associationViewModel.selectedAssociation.collectAsState()
 
-  if (association == null) {
-    Log.e("EditAssociationScreen", "Association not found.")
+  if (association == null && !isNewAssociation) {
+    Log.e("SaveAssociationScreen", "Association not found.")
     Toast.makeText(context, "An error occurred.", Toast.LENGTH_SHORT).show()
     return
   }
 
-  EditAssociationScaffold(
-      association = association!!,
-      onCancel = {
-        associationViewModel.selectAssociation(association!!.uid)
-        navigationAction.navigateTo(Screen.ASSOCIATION_PROFILE, Screen.EDIT_ASSOCIATION)
-      },
-      onSave = { newAssociation ->
-        associationViewModel.saveAssociation(
-            newAssociation,
-            null,
-            onSuccess = {
-              associationViewModel.selectAssociation(newAssociation.uid)
-              navigationAction.navigateTo(Screen.ASSOCIATION_PROFILE, Screen.EDIT_ASSOCIATION)
-            },
-            onFailure = {
-              Log.e("EditAssociationScreen", "Failed to save association.")
-              ToastUtils.showToast(context, context.getString(R.string.save_failed_message))
-            })
+  (if (association == null) {
+        Association(
+            uid =
+                Firebase.firestore
```

Comment:
You should call the repository or the view model and call a method you created to do this instead of doing this in the UI